web/src/buddy_list.js

```diff
@@ -83,12 +88,45 @@ export class BuddyList extends BuddyListConf {
         const items = this.get_data_from_keys({
             keys: more_keys,
         });
+        const subscribed_users = [];
+        const other_users = [];
+        const current_sub = narrow_state.stream_sub();
+        const pm_ids = new Set(narrow_state.pm_ids_string()?.split(","));
+
+        for (const item of items) {
+            if (buddy_data.user_matches_narrow(item.user_id, current_sub, pm_ids)) {
+                subscribed_users.push(item);
+                this.narrow_user_ids.push(item.user_id);
+            } else {
+                other_users.push(item);
+            }
+        }
+
+        const subscribed_users_html = this.items_to_html({
+            items: subscribed_users,
+        });
+        this.$narrow_users_container = $(this.narrow_container_sel);
+        this.$narrow_users_container.append(subscribed_users_html);
 
-        const html = this.items_to_html({
-            items,
+        const other_users_html = this.items_to_html({
+            items: other_users,
         });
-        this.$container = $(this.container_sel);
-        this.$container.append(html);
+        this.$other_users_container = $(this.other_container_sel);
+        this.$other_users_container.append(other_users_html);
+
+        const subscribed_users_is_empty = this.$narrow_users_container.children().length === 0;
+        $("#narrow-users-presence-container").toggleClass("no-display", subscribed_users_is_empty);
+        $("#other-users-presence-container .buddy-list-subsection-header").toggleClass(
+            "no-display",
+            subscribed_users.length === 0,
+        );
+
+        const other_users_is_empty = this.$other_users_container.children().length === 0;
+        $("#other-users-presence-container").toggleClass("no-display", other_users_is_empty);
+        $("#narrow-users-presence-container .buddy-list-subsection-header").toggleClass(
+            "no-display",
+            other_users.length === 0,
+        );
```

Comment:
The subsection headers logic is a bit puzzling to me -- we toggle the other subsection based on it whether this one has rows under it?

Reply:
I made it so that the headers go away if only one section is visible. 

We definitely should do this for Other Users (since having only one section and calling it "other" is odd). 

I'm open to keeping the narrow users header when other users isn't visible, if folks prefer that.